Unity/Assets/Project/Scripts/MyMultiTouchObserver.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace SagoTouchAssignment.MultiTouch {
+
+	using SagoTouch;
+	using System.Collections.Generic;
+	using UnityEngine;
+	using Touch = SagoTouch.Touch;
+
+	public class MyMultiTouchObserver : MonoBehaviour, ISingleTouchObserver {
+
+
+		#region Fields
+
+		[System.NonSerialized]
+		private Camera m_Camera;
+
+		[System.NonSerialized]
+		private Renderer m_Renderer;
+
+		[System.NonSerialized]
+		private List<Touch> m_Touches;
+
+		[System.NonSerialized]
+		private Transform m_Transform;
+
+		#endregion
+
+
+		#region Properties
+
+		public Camera Camera {
+			get { return m_Camera = m_Camera ?? CameraUtils.FindRootCamera(this.Transform); }

```

Comment:
With `UnityEngine.Object` derived objects, you should not use the null-coalescing operator.  They are managed objects, so can be non-null, but still invalid.  Instead you can use the implicit bool operator with a ternary `? :` operator, e.g.
```C#
m_Camera = m_Camera ? m_Camera : CameraUtils.FindRootCamera(this.Transform);
return m_Camera;
```

https://docs.unity3d.com/2018.4/Documentation/ScriptReference/Object.html